jupyterhub_singleuser_profiles/sizes.py

```diff
@@ -26,23 +67,21 @@ def get_form(self, last_size=None):
         """
 
         options = ["<option value='None'>Default (resources will be set based on profiles configured by administrator)</option>"]
-        for size in self.sizes: 
+        for size_i in self.sizes: 
+            size = self._parse(size_i)
+            if not size:
+                continue
+
             selected = ""
             if size['name'] == last_size:
                 selected = "selected=selected"
-            if size['resources'].get('mem_limit') or size['resources'].get('cpu_limit'):
-                # Kept for backwards compatibility
-                options.append("<option value='%s' %s>%s</option>" % (size['name'], selected, "%s (Memory: %s, CPU: %s)" % (size['name'], size['resources']['mem_limit'], size['resources']['cpu_limit'])))
-            else:
-                if size['resources'].get('requests') is None:
-                    size['resources']['requests'] = size['resources']['limits']
-                elif size['resources'].get('limits') is None:
-                    size['resources']['limits'] = size['resources']['requests']
-
-                req_mem = size['resources']['requests']['memory']
-                req_cpu = size['resources']['requests']['cpu']
-                lim_mem = size['resources']['limits']['memory']
-                lim_cpu = size['resources']['limits']['cpu']
-                options.append("<option value='%s' %s>%s</option>" % (size['name'], selected, "%s (Memory request: %s, CPU request: %s, Memory limit: %s, CPU limit: %s)" % (size['name'], req_mem, req_cpu, lim_mem, lim_cpu)))
+
+            req_mem = size['resources']['requests'].get('memory', 'Default')
```

Comment:
It failed on this line when I tried an edge case: 
```sizes:
  - name: Small
    resources:
       requests:
       limits:```

Other than that LGTM

Reply:
Ok, fixed that one, thanks! Merging